### ISSUE TITLE
Combined list of pet issues

### ISSUE THREAD

**Issue by Rokvach** [Regalis11/Barotrauma#4085]
- [x] Pets happiness indicator vibrates when the sub moves

- [x] `Killmonsters` command kills all the pets too

- [x] Playing with a pet in multiplayer doesn't make it jump like it does on singleplayer

- [x] If a pet follows the owner in to the outpost and you undock from the outpost the pets will be left behind and disappear and wont spawn in the submarine like bots. (Tested in singleplayer)

Repro steps: 
1. Acquire couple pets
2. Save game while in a outpost level
3. Load save
4. Pets should now be spawned inside the outpost
5. Undock from outpost and see that the pets which were roaming the outpost are now gone

- [x] Pets disappear when you enter a outpost level, do they have any defined spawn points in outpost levels? (Doesn't happen in multiplayer)

- [x] Pets disappear if you save and quit a campaign in multiplayer

Repro steps:
1. Buy a pet egg from the outpost store
2. Undock from outpost
3. Hatch the pet
4. Save and quit
5. Notice that the pet is gone, in singleplayer the pet is turned back to the egg form like it was on the start of the round

- [x] Pet happiness indicator doesn't show up in multiplayer

- [x] Intensity music starts playing when you hatch a pet in singleplayer, doesn't happen in multiplayer

- [ ] Cthulhu pet keeps constantly dropping its food to the ground when its eating it on multiplayer, picks it up properly in singleplayer

- [x] Game crashes if you spawn in a "Strange Egg" and feed it saline, seems to still use a "smallcrawler"
![image.png](https://images.zenhubusercontent.com/5c924117534a9b0a86cdddce/1284f30d-5c10-4b0b-ba50-21a0a540bdf1)

```
Barotrauma Client crash report (generated on 15.10.2020 12:10:38)


Barotrauma seems to have crashed. Sorry for the inconvenience! 


A0744B4FF8F6D20663028197271CD814


Game version 0.10.6.1 (ReleaseWindows, branch dev, revision d22f22c9c9)
Graphics mode: 2560x1440 (BorderlessWindowed)
VSync OFF
Language: English
Selected content packages: Vanilla 0.9
Level seed: AnaNew Iapetus
Loaded submarine: Dugong (5F3D09DBF368D6EDB69E389EA790999C)
Selected screen: Barotrauma.GameScreen
SteamManager initialized


System info:
    Operating system: Microsoft Windows NT 10.0.18362.0 64 bit
    GPU name: NVIDIA GeForce GTX 980 Ti
    Display mode: {Width:2560 Height:1440 Format:Color AspectRatio:1,7777778}
    GPU status: Normal


Exception: Object reference not set to an instance of an object. (System.NullReferenceException)
Target site: Void .ctor(System.String, Microsoft.Xna.Framework.Vector2, System.String, Barotrauma.CharacterInfo, Boolean, Barotrauma.RagdollParams)
Stack trace: 
   at Barotrauma.Character..ctor(String speciesName, Vector2 position, String seed, CharacterInfo characterInfo, Boolean isRemotePlayer, RagdollParams ragdollParams) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\Character.cs:line 844
   at Barotrauma.Character.Create(String speciesName, Vector2 position, String seed, CharacterInfo characterInfo, Boolean isRemotePlayer, Boolean hasAi, Boolean createNetworkEvent, RagdollParams ragdoll) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\Character.cs:line 823
   at Barotrauma.EntitySpawner.CharacterSpawnInfo.Spawn() in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Networking\EntitySpawner.cs:line 112
   at Barotrauma.EntitySpawner.Update() in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Networking\EntitySpawner.cs:line 301
   at Barotrauma.MapEntity.UpdateAll(Single deltaTime, Camera cam) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Map\MapEntity.cs:line 524
   at Barotrauma.GameScreen.Update(Double deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Screens\GameScreen.cs:line 228
   at Barotrauma.GameMain.Update(GameTime gameTime) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\GameMain.cs:line 921
   at Microsoft.Xna.Framework.Game.DoUpdate(GameTime gameTime) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 656
   at Microsoft.Xna.Framework.Game.Tick() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 500
   at Microsoft.Xna.Framework.SdlGamePlatform.RunLoop() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\SDL\SDLGamePlatform.cs:line 92
   at Microsoft.Xna.Framework.Game.Run(GameRunBehavior runBehavior) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 397
   at Microsoft.Xna.Framework.Game.Run() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 367
   at Barotrauma.Program.Main(String[] args) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Program.cs:line 58


Last debug messages:
[15.10.2020 12:10:36] spawnitem "Strange Egg" inventory
[15.10.2020 12:09:19] Generating an outpost for the start of the level... (Location: Ana Station, level type: LocationConnection)
[15.10.2020 12:09:10] Transitioning to AnaNew Iapetus (current location: Ana Station, selected location: New Iapetus, leaving sub: Dugong, at start: True, at end: False, transition type: LeaveLocation)
[15.10.2020 12:08:58] control "Masako Lawrence"
[15.10.2020 12:08:31] freecamera
[15.10.2020 12:08:19] Generating an outpost for the start of the level... (Location: Ana Station, level type: Outpost)
[15.10.2020 12:08:11] Transitioning to Ana (current location: The city of The Bucket, selected location: Ana Station, leaving sub: Dugong, at start: False, at end: True, transition type: ProgressToNextLocation)
[15.10.2020 12:07:54] teleportsub end
[15.10.2020 12:05:07] Generating an outpost for the end of the level... (Location: Ana Station, level type: LocationConnection)
[15.10.2020 12:05:07] Generating an outpost for the start of the level... (Location: The city of The Bucket, level type: LocationConnection)
[15.10.2020 12:03:31] Lighting disabled
[15.10.2020 12:03:31] lighting
[15.10.2020 12:03:18] Generating an outpost for the end of the level... (Location: Ana Station, level type: LocationConnection)
[15.10.2020 12:03:18] Generating an outpost for the start of the level... (Location: The city of The Bucket, level type: LocationConnection)
[15.10.2020 12:03:09] Transitioning to The BucketAna (current location: The city of The Bucket, selected location: Ana Station, leaving sub: Dugong, at start: True, at end: False, transition type: LeaveLocation)
[15.10.2020 12:02:36] money 10000
[15.10.2020 12:02:36] money 10000
[15.10.2020 12:02:35] money 10000
[15.10.2020 12:02:35] money 10000
[15.10.2020 12:02:35] money 10000
[15.10.2020 12:02:35] money 10000
[15.10.2020 12:02:34] money 10000
[15.10.2020 12:02:32] Steam achievements have been disabled during this play session.
[15.10.2020 12:02:32] Enabled cheat commands.
[15.10.2020 12:02:32] enablecheats
[15.10.2020 12:02:12] Generating an outpost for the start of the level... (Location: The city of The Bucket, level type: Outpost)
[15.10.2020 12:02:07] WARNING: GameSession set to null while in the game screen
   at System.Environment.get_StackTrace()
   at Barotrauma.GameMain.set_GameSession(GameSession value) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\GameMain.cs:line 82
   at Barotrauma.Networking.GameClient.Disconnect() in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Networking\GameClient.cs:line 2609
   at Barotrauma.Networking.GameClient.CancelConnect() in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Networking\GameClient.cs:line 432
   at Barotrauma.Networking.GameClient.OnDisconnect(Boolean disableReconnect) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Networking\GameClient.cs:line 1008
   at Barotrauma.Networking.SteamP2POwnerPeer.Close(String msg, Boolean disableReconnect) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Networking\Primitives\Peers\SteamP2POwnerPeer.cs:line 418
   at Barotrauma.Networking.GameClient.Disconnect() in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Networking\GameClient.cs:line 2575
   at Barotrauma.GameMain.QuitToMainMenu(Boolean save) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\GameMain.cs:line 1062
   at Barotrauma.GUI.<>c.<TogglePauseMenu>b__181_15(GUIButton yesBtn, Object userdata) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\GUI\GUI.cs:line 2243
   at Barotrauma.GUIButton.Update(Single deltaTime) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\GUI\GUIButton.cs:line 255
   at System.Collections.Generic.List`1.ForEach(Action`1 action)
   at Barotrauma.GUI.Update(Single deltaTime) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\GUI\GUI.cs:line 1212
   at Barotrauma.GameMain.Update(GameTime gameTime) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\GameMain.cs:line 948
   at Microsoft.Xna.Framework.Game.DoUpdate(GameTime gameTime) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 651
   at Microsoft.Xna.Framework.Game.Tick() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 504
   at Microsoft.Xna.Framework.SdlGamePlatform.RunLoop() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\SDL\SDLGamePlatform.cs:line 94
   at Microsoft.Xna.Framework.Game.Run(GameRunBehavior runBehavior) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 398
   at Microsoft.Xna.Framework.Game.Run() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 368
   at Barotrauma.Program.Main(String[] args) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Program.cs:line 59
[15.10.2020 12:01:16] Rokvach
[15.10.2020 12:01:14] Task count: 0
[15.10.2020 12:01:13] Generating an outpost for the end of the level... (Location: The city of Dank, level type: LocationConnection)
[15.10.2020 12:00:47] Rokvach
[15.10.2020 11:59:53] Server command: spawnitem Saline inventory
[15.10.2020 11:59:53] spawnitem Saline inventory
[15.10.2020 11:59:53] Server command: spawnitem Saline inventory
[15.10.2020 11:59:53] spawnitem Saline inventory
[15.10.2020 11:59:53] Server command: spawnitem Saline inventory
[15.10.2020 11:59:53] spawnitem Saline inventory
[15.10.2020 11:59:52] Server command: spawnitem Saline inventory
[15.10.2020 11:59:52] spawnitem Saline inventory
[15.10.2020 11:59:52] Server command: spawnitem Saline inventory
[15.10.2020 11:59:52] spawnitem Saline inventory
[15.10.2020 11:58:59] Lighting disabled
[15.10.2020 11:58:59] lighting
[15.10.2020 11:58:43] Rokvach
[15.10.2020 11:58:41] Task count: 0
[15.10.2020 11:58:39] Generating an outpost for the end of the level... (Location: Hyperenor Linea, level type: LocationConnection)
[15.10.2020 11:58:29] Rokvach
[15.10.2020 11:58:19] Server command: teleportsub end
[15.10.2020 11:58:19] teleportsub end
[15.10.2020 11:58:17] Server command: spawnitem Saline inventory
[15.10.2020 11:58:17] spawnitem Saline inventory
[15.10.2020 11:56:31] Server command: spawnitem Saline inventory
[15.10.2020 11:56:31] spawnitem Saline inventory
[15.10.2020 11:56:30] Server command: spawnitem Saline inventory
[15.10.2020 11:56:30] spawnitem Saline inventory
[15.10.2020 11:56:30] Server command: spawnitem Saline inventory
[15.10.2020 11:56:30] spawnitem Saline inventory
[15.10.2020 11:56:30] Server command: spawnitem Saline inventory
[15.10.2020 11:56:30] spawnitem Saline inventory
[15.10.2020 11:56:30] Server command: spawnitem Saline inventory
[15.10.2020 11:56:30] spawnitem Saline inventory
[15.10.2020 11:56:29] Server command: spawnitem Saline inventory
[15.10.2020 11:56:29] spawnitem Saline inventory
[15.10.2020 11:56:29] Server command: spawnitem Saline inventory
[15.10.2020 11:56:29] spawnitem Saline inventory
[15.10.2020 11:56:25] Server command: teleportcharacter Rokvach
[15.10.2020 11:56:25] teleportcharacter Rokvach
[15.10.2020 11:56:14] Rokvach
[15.10.2020 11:56:12] Task count: 0
[15.10.2020 11:56:00] Rokvach
[15.10.2020 11:55:49] Server command: teleportsub end
[15.10.2020 11:55:49] teleportsub end
[15.10.2020 11:55:19] Rokvach
[15.10.2020 11:55:19] Server command: control Rokvach
[15.10.2020 11:55:19] control Rokvach
[15.10.2020 11:55:17] Server command: teleportcharacter Rokvach
[15.10.2020 11:55:17] teleportcharacter Rokvach
[15.10.2020 11:55:10] Rokvach
[15.10.2020 11:55:10] freecamera
[15.10.2020 11:54:00] Server command: spawnitem Saline inventory
[15.10.2020 11:54:00] spawnitem Saline inventory
[15.10.2020 11:54:00] Server command: spawnitem Saline inventory
[15.10.2020 11:54:00] spawnitem Saline inventory
[15.10.2020 11:54:00] Server command: spawnitem Saline inventory
[15.10.2020 11:54:00] spawnitem Saline inventory
[15.10.2020 11:54:00] Server command: spawnitem Saline inventory
[15.10.2020 11:54:00] spawnitem Saline inventory
[15.10.2020 11:53:59] Server command: spawnitem Saline inventory
[15.10.2020 11:53:59] spawnitem Saline inventory
[15.10.2020 11:53:59] Server command: spawnitem Saline inventory
[15.10.2020 11:53:59] spawnitem Saline inventory
[15.10.2020 11:53:58] Server command: spawnitem Saline inventory
[15.10.2020 11:53:58] spawnitem Saline inventory
[15.10.2020 11:51:38] Rokvach
[15.10.2020 11:51:37] Task count: 0
[15.10.2020 11:51:35] Generating an outpost for the start of the level... (Location: Newellstead Station, level type: LocationConnection)
[15.10.2020 11:51:24] Rokvach
[15.10.2020 11:51:24] WARNING: Could not determine reputation change for faction "Children of the Honkmother" (faction was not present at the start of the round).
[15.10.2020 11:51:24] WARNING: Could not determine reputation change for faction "The Church of Husk" (faction was not present at the start of the round).
[15.10.2020 11:51:24] WARNING: Could not determine reputation change for faction "Jovian Separatists" (faction was not present at the start of the round).
[15.10.2020 11:51:24] WARNING: Could not determine reputation change for faction "Europa Coalition" (faction was not present at the start of the round).
[15.10.2020 11:50:18] Server command: money 10000
[15.10.2020 11:50:18] money 10000
[15.10.2020 11:50:18] Server command: money 10000
[15.10.2020 11:50:18] money 10000
[15.10.2020 11:50:18] Server command: money 10000
[15.10.2020 11:50:18] money 10000
[15.10.2020 11:50:18] Server command: money 10000
[15.10.2020 11:50:18] money 10000
[15.10.2020 11:50:17] Server command: money 10000
[15.10.2020 11:50:17] money 10000
[15.10.2020 11:50:17] Server command: money 10000
[15.10.2020 11:50:17] money 10000
[15.10.2020 11:50:17] Server command: money 10000
[15.10.2020 11:50:17] money 10000
[15.10.2020 11:50:14] Server command: enablecheats
[15.10.2020 11:50:14] enablecheats
[15.10.2020 11:49:49] Rokvach
[15.10.2020 11:49:47] Task count: 0
[15.10.2020 11:49:45] Generating an outpost for the start of the level... (Location: Newellstead Station, level type: Outpost)
[15.10.2020 11:49:19] Lobby created!
[15.10.2020 11:49:06] (Msg) RelayNetWorkStatus:  avail=OK  config=OK  anyrelay=OK   (OK)
[15.10.2020 11:49:06] (Msg) Ping location: sto2=13+1,sto=14+1/13+1,fra=39+3/32+1,ams=38+3/33+1,waw=43+4/34+1,lux=46+4/36+1,lhr=45+4/37+1,par=50+5/42+1,vie=50+5/45+1,iad=127+12/118+1,sgp=199+19/200+19,gru=256+25/262+1
[15.10.2020 11:49:06] (Msg) Ping measurement completed
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'syd' established.
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'tyo1' established.
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'tyo' established.
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'maa' established.
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'sgp' established.
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'sea' established.
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'lim' established.
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'scl' established.
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'lax' established.
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'okc' established.
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'vie' established.
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'gru' established.
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'waw' established.
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'sof' established.
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'ord' established.
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'hkg' established.
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'jnb' established.
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'sto' established.
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'sto2' established.
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'eat' established.
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'mad' established.
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'bom' established.
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'par' established.
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'dxb' established.
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'lux' established.
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'lhr' established.
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'atl' established.
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'iad' established.
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'fra' established.
[15.10.2020 11:49:01] (Verbose) Communications with routing cluster 'ams' established.
[15.10.2020 11:49:01] (Msg) RelayNetWorkStatus:  avail=Attempting  config=OK  anyrelay=Attempting   (Performing ping measurement)
[15.10.2020 11:49:01] (Msg) Performing ping measurement
[15.10.2020 11:49:01] (Msg) Got SDR network config.  Loaded revision 283 OK
[15.10.2020 11:49:01] (Warning) SDR network config fetch first attempt failed.  HTTP 504.  .  Trying again.
[15.10.2020 11:48:53] Attempting to open ALC device "OpenAL Soft on Kaiuttimet (Realtek High Definition Audio)"
[15.10.2020 11:48:52] Loaded languages: English, German, French, Russian, Polish, Brazilian Portuguese, Castilian Spanish, Latinamerican Spanish, Simplified Chinese, Traditional Chinese, Japanese, Turkish, Korean
[15.10.2020 11:48:52] Loaded languages: English, German, French, Russian, Polish, Brazilian Portuguese, Castilian Spanish, Latinamerican Spanish, Simplified Chinese, Traditional Chinese, Japanese, Turkish, Korean
```

**Comment by Rokvach** [comment]
Added couple new issues at the top of the list and tested the already fixed issues, all of the currently fixed ones are working correctly.

**Comment by itchyOwl** [comment]
"killmonsters" issue fixed in Regalis11/Barotrauma-development@07f17fb (bugfixes).

**Comment by batturing** [comment]
I'm not sure if this is the right place to post this so I'm sorry if it isn't. I'm having a problem with the cthulhu pet where it seems to be clipping into the walls of the sub or something because it keeps slamming into the walls and floor and dying, even when the sub isn't moving. I am playing in singleplayer currently, not sure if this happens in multiplayer. Grabbing or playing with them doesn't seem to fix it. Grabbing and holding them stops it temporarily but once you let them go again they continue slamming into walls and dying. I couldn't find anything about this posted anywhere.
